sdk/highlight-go/highlight.go

```diff
@@ -247,12 +249,15 @@ func InterceptRequest(r *http.Request) context.Context {
 // for a particular request from the request headers, adding the values to the provided context.
 func InterceptRequestWithContext(ctx context.Context, r *http.Request) context.Context {
 	highlightReqDetails := r.Header.Get("X-Highlight-Request")
+
 	ids := strings.Split(highlightReqDetails, "/")
-	if len(ids) < 2 {
-		return ctx
+	if len(ids) >= 2 {
+		ctx = context.WithValue(ctx, ContextKeys.SessionSecureID, ids[0])
+		ctx = context.WithValue(ctx, ContextKeys.RequestID, ids[1])
+	} else {
+		ctx = otel.GetTextMapPropagator().Extract(ctx, propagation.HeaderCarrier(r.Header))
```

Comment:
does this need to happen in any other backend sdks?

Reply:
Don't believe so. Same reasoning as https://github.com/highlight/highlight/pull/8706#discussion_r1646675080, but let me know if you think I'm off there.

Reply:
i think this does affect other customers tho since they may want to propagate the session ID to the backend trace, so their backend trace is associated to the session. maybe we can test that with an e2e app (like a customer react + python opentelemetry native no highlight)